### PR TITLE
fix: display wrong collateral value

### PR DESCRIPTION
**Pull request by maikelordaz**
The return value from getWithdrawableUserBalance takes in consideration only the collateral state, now also check if the user have paid for the cycle.

Also was changed the flow of withdrawCollateral to use getWithdrawableUserBalace, as it was the same logic to calculate the withdraw amount

### CODE REVIEW

File: contracts/facets/CollateralFacet.sol [hunk 1]
```diff
@@ -270,36 +273,28 @@ contract CollateralFacet is ICollateral {
...
-            uint minRequiredCollateral = (IGetters(address(this)).getRemainingCyclesContributionWei(
-                _termId
-            ) * 15) / 10; // 1.5 X RCC in wei
+            collateral.collateralMembersBank[msg.sender] -= allowedWithdrawal;
 
-            // Collateral must be higher than 1.5 X RCC
-            if (userCollateral > minRequiredCollateral) {
-                uint allowedWithdrawal = userCollateral - minRequiredCollateral; // We allow to withdraw the positive difference
-                collateral.collateralMembersBank[msg.sender] -= allowedWithdrawal;
+            _withdrawFromYield(_termId, msg.sender, allowedWithdrawal, yield);
```
Comment (MohammedTaka) [inline contracts/facets/CollateralFacet.sol:293]: Tabs here seem wrong

File: contracts/facets/GettersFacet.sol [hunk 2]
```diff
@@ -311,8 +312,22 @@ contract GettersFacet is IGetters {
...
             allowedWithdrawal = userCollateral + availableYield;
         } else if (collateral.state == LibCollateralStorage.CollateralStates.CycleOngoing) {
-            // Everything above 1.5 X remaining cycles contribution (RCC) can be withdrawn
-            uint minRequiredCollateral = (getRemainingCyclesContributionWei(termId) * 15) / 10; // 1.5 X RCC in wei
+            // uint minRequiredCollateral = (getRemainingCyclesContributionWei(termId) * 15) / 10; // 1.5 X RCC in wei
+
+            uint minRequiredCollateral;
+
+            // Check if the user has paid this cycle
+            if (!fund.paidThisCycle[user]) {
```
Comment (MohammedTaka) [inline contracts/facets/GettersFacet.sol:318]: Wouldn't this be better to put in the getRemainingCyclesContributionWei function?

File: contracts/facets/GettersFacet.sol [hunk 3]
```diff
@@ -311,8 +312,22 @@ contract GettersFacet is IGetters {
...
+                // Everything above 1.5 X remaining cycles contribution (RCC) can be withdrawn
+                minRequiredCollateral = (getRemainingCyclesContributionWei(termId) * 15) / 10; // 1.5 X RCC in wei
+            } else {
+                uint remainingCycles = fund.totalAmountOfCycles - fund.currentCycle;
+                uint contributionAmountWei = getToCollateralConversionRate(
+                    term.contributionAmount * 10 ** 18
+                );
+
+                minRequiredCollateral = remainingCycles * contributionAmountWei;
```
Comment (MohammedTaka) [inline contracts/facets/GettersFacet.sol]: Missing * 15 / 10 here

**Review by MohammedTaka — CHANGES_REQUESTED** [review]
Logic

File: contracts/facets/GettersFacet.sol (hunk 2 above)
Comment (maikelordaz) [reply contracts/facets/GettersFacet.sol:318]: You mean for the `minRequiresCollateral`? I used it if the user have not paid, if the user have already paid the cycle I'm calculating the amount with one cycle less

File: contracts/facets/GettersFacet.sol (hunk 3 above)
Comment (maikelordaz) [reply contracts/facets/GettersFacet.sol]: Right. Commit 07c3649

**Review by MohammedTaka — APPROVED** [review]

**Pull request merged**